sdks/java/core/src/main/java/org/apache/beam/sdk/coders/AtomicCoder.java

```diff
@@ -35,11 +35,18 @@
   /**
    * {@inheritDoc}.
    *
-   * @throws NonDeterministicException
+   * <p>By default, does not throw.
```

Comment:
Suggestion: "unless overridden, does not throw: an AtomicCoder is assumed to be deterministic"